packages/story-editor/src/components/panels/document/taxonomies/FlatTermSelector.js

```diff
@@ -87,10 +96,19 @@ function FlatTermSelector({ taxonomy, canCreateTerms }) {
   );
 
   const handleFreeformInputChange = useDebouncedCallback((value) => {
+    if (value.length === 0) {
+      setFlatSearchResults([]);
+      return;
+    }
+
     if (value.length < 3) {
       return;
     }
-    addSearchResultsToCache(taxonomy, { name: value });
+    addSearchResultsToCache(taxonomy, {
+      search: value,
+      // This is the per_page value Gutenberg is using
+      per_page: 20,
+    });
```

Comment:
after you move the `searchResults` to be encapsulated in this component vs the provider, you should be able to just do this here:
```
const results = await  addSearchResultsToCache(taxonomy, {
      search: value,
      // This is the per_page value Gutenberg is using
      per_page: 20,
});
setSearchResults(results);
```